### ISSUE TITLE
java.util.NoSuchElementException: None.get slick.compiler.FuseComprehensions.select(Relational.scala:432)

### ISSUE THREAD

**Issue by TimothyKlim** [slick/slick#1137]
Hi!
I'm trying to join multiple tables with Slick 3.0:
```scala
class UserTable(tag: Tag) extends Table[models.User](tag, "users") {
  def id = column[UUID]("id", O.PrimaryKey)
  def email = column[String]("email")
  def firstName = column[Option[String]]("first_name")

  def * = (id, email, firstName) <> (models.User.tupled, models.User.unappy)
}

class RoleTable(tag: Tag) extends Table[models.Role](tag, "roles") {
  def id = column[UUID]("id", O.PrimaryKey)
  def name = column[String]("name")
  def rights = column[List[RoleType]]("rights")

  def * = (id, name, rights) <> (models.Role.tupled, models.Role.unappy)
}

class RoleUserTable(tag: Tag) extends Table[RoleUser](tag, "roles_users") {
  def roleId = column[UUID]("role_id", O.PrimaryKey)
  def userId = column[UUID]("user_id", O.PrimaryKey)

  def * = (roleId, userId) <> (RoleUser.tupled, RoleUser.unapply)
}

  val arrayAggCustom =
    SimpleFunction.unary[List[RoleType.RoleType], List[RoleType.RoleType]]("array_agg_custom")

  val table = TableQuery[UserTable]
  val tableWithNested = table
    .joinLeft(RoleUser.table).on(_.id === _.userId)
    .joinLeft(Role.table).on(_._2.map(_.roleId) === _.id)
    .map { t => (t._1, t._2.map { x => arrayAggCustom(x.rights) }) }
    .groupBy { x => (x._1, x._2) }
    .map(_._1)
```
and I got this error:
```
e: java.util.NoSuchElementException: None.get
scala.None$.get(Option.scala:347)
scala.None$.get(Option.scala:345)
slick.compiler.FuseComprehensions.select(Relational.scala:432)
slick.compiler.FuseComprehensions.select(Relational.scala:434)
slick.compiler.ConvertToComprehensions$$anonfun$4.applyOrElse(Relational.scala:126)
slick.compiler.ConvertToComprehensions$$anonfun$4.applyOrElse(Relational.scala:113)
slick.ast.NodeOps$.replace(Util.scala:94)
slick.ast.NodeOps$.replace$extension(Util.scala:43)
slick.ast.NodeOps$$anonfun$replace$1$$anonfun$applyOrElse$1.apply(Util.scala:94)
slick.ast.NodeOps$$anonfun$replace$1$$anonfun$applyOrElse$1.apply(Util.scala:94)
slick.ast.Util$$anonfun$mapOrNone$1.apply(Util.scala:15)
slick.ast.Util$$anonfun$mapOrNone$1.apply(Util.scala:14)
scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
slick.ast.Util$.mapOrNone(Util.scala:14)
slick.ast.Node$class.nodeMapChildren(Node.scala:44)
slick.ast.StructNode.nodeMapChildren(Node.scala:197)
slick.ast.NodeOps$$anonfun$replace$1.applyOrElse(Util.scala:94)
slick.ast.NodeOps$$anonfun$replace$1.applyOrElse(Util.scala:94)
scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
slick.compiler.ConvertToComprehensions$$anonfun$4.applyOrElse(Relational.scala:113)
slick.compiler.ConvertToComprehensions$$anonfun$4.applyOrElse(Relational.scala:113)
slick.ast.NodeOps$.replace(Util.scala:94)
slick.ast.NodeOps$.replace$extension(Util.scala:43)
slick.compiler.ConvertToComprehensions.convert1(Relational.scala:113)
slick.compiler.ConvertToComprehensions.convert(Relational.scala:76)
slick.compiler.ConvertToComprehensions$$anonfun$8.apply(Relational.scala:76)
slick.compiler.ConvertToComprehensions$$anonfun$8.apply(Relational.scala:76)
slick.ast.Util$$anonfun$mapOrNone$1.apply(Util.scala:15)
slick.ast.Util$$anonfun$mapOrNone$1.apply(Util.scala:14)
scala.collection.immutable.List.foreach(List.scala:381)
slick.ast.Util$.mapOrNone(Util.scala:14)
slick.ast.Node$class.nodeMapChildren(Node.scala:44)
slick.ast.Bind.nodeMapChildren(Node.scala:446)
slick.compiler.ConvertToComprehensions.convert(Relational.scala:76)
slick.compiler.ConvertToComprehensions$$anonfun$apply$2$$anonfun$apply$3.apply(Relational.scala:69)
slick.compiler.ConvertToComprehensions$$anonfun$apply$2$$anonfun$apply$3.apply(Relational.scala:69)
slick.ast.ClientSideOp$.mapServerSide(ClientSideOp.scala:20)
slick.ast.ClientSideOp$$anonfun$mapServerSide$1.apply(ClientSideOp.scala:19)
slick.ast.ClientSideOp$$anonfun$mapServerSide$1.apply(ClientSideOp.scala:19)
slick.ast.ResultSetMapping$$anonfun$1.apply(ClientSideOp.scala:66)
slick.ast.ResultSetMapping$$anonfun$1.apply(ClientSideOp.scala:65)
scala.Function2$$anonfun$tupled$1.apply(Function2.scala:54)
scala.Function2$$anonfun$tupled$1.apply(Function2.scala:53)
scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
scala.collection.Iterator$class.foreach(Iterator.scala:750)
scala.collection.AbstractIterator.foreach(Iterator.scala:1202)
scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
scala.collection.AbstractIterable.foreach(Iterable.scala:54)
scala.collection.TraversableLike$class.map(TraversableLike.scala:245)
scala.collection.AbstractTraversable.map(Traversable.scala:104)
slick.ast.DefNode$class.nodeMapScopedChildren(Symbol.scala:55)
slick.ast.ResultSetMapping.nodeMapScopedChildren(ClientSideOp.scala:43)
slick.ast.ResultSetMapping.nodeMapServerSide(ClientSideOp.scala:65)
slick.ast.ClientSideOp$.mapServerSide(ClientSideOp.scala:19)
slick.compiler.ConvertToComprehensions$$anonfun$apply$2.apply(Relational.scala:69)
slick.compiler.ConvertToComprehensions$$anonfun$apply$2.apply(Relational.scala:69)
slick.compiler.CompilerState.map(QueryCompiler.scala:181)
slick.compiler.ConvertToComprehensions.apply(Relational.scala:69)
slick.compiler.QueryCompiler$$anonfun$runPhase$1.apply(QueryCompiler.scala:79)
slick.compiler.QueryCompiler$$anonfun$runPhase$1.apply(QueryCompiler.scala:78)
scala.util.DynamicVariable.withValue(DynamicVariable.scala:58)
slick.ast.SymbolNamer.use(Symbol.scala:92)
slick.compiler.QueryCompiler.runPhase(QueryCompiler.scala:78)
slick.compiler.QueryCompiler$$anonfun$3.apply(QueryCompiler.scala:65)
slick.compiler.QueryCompiler$$anonfun$3.apply(QueryCompiler.scala:63)
scala.collection.TraversableOnce$$anonfun$foldLeft$1.apply(TraversableOnce.scala:142)
scala.collection.TraversableOnce$$anonfun$foldLeft$1.apply(TraversableOnce.scala:142)
scala.collection.Iterator$class.foreach(Iterator.scala:750)
scala.collection.AbstractIterator.foreach(Iterator.scala:1202)
scala.collection.TraversableOnce$class.foldLeft(TraversableOnce.scala:142)
scala.collection.AbstractIterator.foldLeft(Iterator.scala:1202)
slick.compiler.QueryCompiler.runPhases(QueryCompiler.scala:63)
slick.compiler.QueryCompiler.run(QueryCompiler.scala:53)
slick.compiler.QueryCompiler.run(QueryCompiler.scala:47)
slick.profile.BasicProfile$API$class.streamableQueryActionExtensionMethods(BasicProfile.scala:71)
...
```

Where am I wrong?

**Comment by TimothyKlim** [comment]
PoC of this bug https://github.com/TimothyKlim/slick3.0-bug-poc

**Comment by szeiger** [comment]
Fixed in master via #1133

**Comment by TimothyKlim** [comment]
@szeiger thank you!
